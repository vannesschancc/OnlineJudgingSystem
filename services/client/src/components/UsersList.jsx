import React from 'react';

/*
Notice how we used props instead of state in this component. Essentially, you can pass state to a component with either props or state:
    Props: data flows down via props (from state to props), read only
    State: data is tied to a component, read and write

*/
const UserList = (probs) => {
    return (
        <div>
            {
                probs.users.map((user) => {
                    return (
                        <h4
                            key={user.id}
                            className='box title is-4'
                        >{user.username} 
                        </h4>
                    )
                })
            }
        </div>
    )
};

export default UserList;