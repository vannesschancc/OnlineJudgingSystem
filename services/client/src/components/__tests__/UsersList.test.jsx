import React from 'react';
import { shallow } from 'enzyme';

import UsersList from '../UsersList';
import renderer from 'react-test-renderer';

const users = [
    {
        'active': true,
        'email': 'vannesschancc@live.com',
        'id': 1,
        'username': 'vannesschancc'
    },
    {
        'active': true,
        'email': 'chc507@ucsd.edu',
        'id': 2,
        'username': 'chc507'
    } 
];

/*
    In this test, we used the shallow helper method to create the UsersList component and then we retrieved the output and made assertions against it. It's important to note that with "shallow rendering", we can test the component in complete isolation, which helps to ensure child components do not indirectly affect assertions.
*/
test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4')
    expect(element.length).toBe(2);
    expect(element.get(0).props.children).toBe('vannesschancc');
});

//test ui, make sure that it does not change
test('UserList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});