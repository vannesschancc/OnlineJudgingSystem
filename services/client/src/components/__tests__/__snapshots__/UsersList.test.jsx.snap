// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    vannesschancc
  </h4>
  <h4
    className="box title is-4"
  >
    chc507
  </h4>
</div>
`;
